packages/vite/src/node/server/moduleGraph.ts

```diff
@@ -60,12 +84,20 @@ export class ModuleGraph {
   // a single file may corresponds to multiple modules with different queries
   fileToModulesMap = new Map<string, Set<ModuleNode>>()
   safeModulesPath = new Set<string>()
+  private saveCacheState: {
+    promise?: Promise<void>
+    timeout?: NodeJS.Timeout
+    canQueue: boolean
+  } = { canQueue: true }
```

Comment:
maybe init canQueue to false if this.config.cacheDir isn't set to avoid triggering unneeded callbacks ?

Reply:
I'm suggesting to remove the problem completely: https://github.com/vitejs/vite/pull/6415

Reply:
I'd advocate for a (another then) way/setting to disable this caching, just in case